src/components/Cards/PostCard.tsx

```diff
@@ -0,0 +1,92 @@
+import { Group, Stack, Text, UnstyledButton } from '@mantine/core';
+import React from 'react';
+import { FeedCard } from '~/components/Cards/FeedCard';
+import { useCardStyles } from '~/components/Cards/Cards.styles';
+import { EdgeImage } from '~/components/EdgeImage/EdgeImage';
+import { ImageGuard } from '~/components/ImageGuard/ImageGuard';
+import { MediaHash } from '~/components/ImageHash/ImageHash';
+import { UserAvatar } from '~/components/UserAvatar/UserAvatar';
+import { PostsInfiniteModel } from '~/server/services/post.service';
+import { useRouter } from 'next/router';
+import { IconPhoto } from '@tabler/icons-react';
+import { abbreviateNumber } from '~/utils/number-helpers';
+import { IconBadge } from '~/components/IconBadge/IconBadge';
+
+const IMAGE_CARD_WIDTH = 332;
+
+export function PostCard({ data }: Props) {
+  const { classes, cx } = useCardStyles();
+  const router = useRouter();
+  return (
+    <FeedCard href={`/posts/${data.id}`} aspectRatio="square">
+      <div className={classes.root}>
+        {data.image && (
+          <ImageGuard
+            images={[data.image]}
+            connect={{ entityId: data.id, entityType: 'post' }}
+            render={(image) => (
+              <ImageGuard.Content>
+                {({ safe }) => (
+                  <>
+                    <ImageGuard.Report context="post" />
+                    <ImageGuard.ToggleConnect position="top-left" />
+                    {!safe ? (
+                      <MediaHash {...data.image} />
+                    ) : (
+                      <EdgeImage
+                        src={image.url}
+                        name={image.name ?? image.id.toString()}
+                        alt={image.name ?? undefined}
+                        width={IMAGE_CARD_WIDTH}
+                        placeholder="empty"
+                        style={{ width: '100%', position: 'relative' }}
+                      />
+                    )}
+                  </>
+                )}
+              </ImageGuard.Content>
+            )}
+          />
+        )}
+        <Stack
+          className={cx(classes.contentOverlay, classes.bottom, classes.gradientOverlay)}
+          spacing="sm"
+        >
+          <Group position="apart" align="end" noWrap>
+            <Stack>
+              {data.user?.id !== -1 && (
+                <UnstyledButton
+                  sx={{ color: 'white' }}
+                  onClick={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+
+                    router.push(`/users/${data.user.username}`);
```

Comment:
Same here, let's link it to their posts tabs
```suggestion
                    router.push(`/users/${data.user.username}/posts`);
```